packages/integration/api-report.md

```diff
@@ -272,6 +284,18 @@ export type GoogleGcsIntegrationConfig = {
   privateKey?: string;
 };
 
+// @public
+export interface IGithubCredentialsProvider {
+  // (undocumented)
+  getCredentials(opts: { url: string }): Promise<GithubCredentials>;
+}
+
+// @public
+export interface IGithubCredentialsProviderFactory {
```

Comment:
How come this extra factory indirection is needed? It feels like it should be possible to implement this indirection within a custom `GithubCredentialsProvider` implementation that selects from multiple configs based on the URL

Reply:
I think it could also be a static function on the GithubCredentialsProvider however static functions of classes cannot be typed. That makes it difficult to pass the reference around. Maybe I am missing something.